Sources/Datadog/DatadogCore.swift

```diff
@@ -0,0 +1,46 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Core implementation of Datadog SDK.
+///
+/// The core provides a storage and upload mechanism for each registered
+/// feature based on their respective configuration.
+///
+/// By complying with `DatadogFeatureRegistry`, the core can
+/// provide context and writing scopes to features for event recording.
+internal final class DatadogCore {
+    /// The user consent provider for collecting PII.
+    let consentProvider: ConsentProvider
```

Comment:
This comment is IMO misleading - tracking consent has nothing to do with PIIs, how about simply:
```suggestion
    /// The user tracking consent provider.
    let consentProvider: ConsentProvider
```